shell/platform/common/text_editing_delta.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
+
+#include <codecvt>
+#include <string>
+
+#include "flutter/shell/platform/common/text_range.h"
+
+namespace flutter {
+
+struct TextEditingDelta {
+ public:
+  TextEditingDelta(std::u16string textBeforeChange,
+                   TextRange range,
+                   std::u16string text);
+  virtual ~TextEditingDelta();
+
+  std::string oldText() const { return utf16ToUtf8(oldText_); }
+  std::string deltaText() const { return utf16ToUtf8(deltaText_); }
+  int deltaStart() const { return deltaStart_; }
+  int deltaEnd() const { return deltaEnd_; }
+
+  TextEditingDelta(std::string textBeforeChange,
```

Comment:
Here and elsewhere: for parameters, use `const std::string& param_name`.

(Yes, we could use `std::string_view` and `std::u16string_view`, but I'd rather hold off and do a pass over the whole codebase where we do something consistent -- for now, similar classes like `TextInputModel` use string refs, so best for us to to keep consistent with them)